modules/mssqlserver/src/test/java/org/testcontainers/junit/mssqlserver/SimpleMSSQLServerTest.java

```diff
@@ -64,4 +65,14 @@ public void testSetupDatabase() throws SQLException {
             assertThat(resultSetInt).as("A basic SELECT query succeeds").isEqualTo(3);
         }
     }
+
+    @Test
+    public void testListeningPortWaitStrategy() throws Exception {
+        try (
+            MSSQLServerContainer<?> mssqlServer = new MSSQLServerContainer<>(MSSQLServerTestImages.MSSQL_SERVER_IMAGE)
+        ) {
+            mssqlServer.start();
+            Wait.forListeningPort().waitUntilReady(mssqlServer);
+        }
+    }
```

Comment:
I am a bit on the fence with this test. Idk if
```
Wait.forListeningPort().waitUntilReady(mssqlServer);
```
was intended to be used outside the context of a container. That being said I did try
```
MSSQLServerContainer<?> mssqlServer = new MSSQLServerContainer<>(MSSQLServerTestImages.MSSQL_SERVER_IMAGE)
    .waitingFor(Wait.forListeningPort())
```
but it ignores this waiting strategy and opts for the one in JdbcDatabaseContainer which I assume is intentional. And I also tried extending MSSQLServerContainer and overriding `waitUntilContainerStarted` which works but it still ends up calling this same method - the above approach had way less boilerplate in comparison. Keen to hear your thoughts.

Reply:
This line is unnecessary, `start()` will already block until waiting condition is reached.

Reply:
If we want to check for the contract of the method, we should just check the getter itself and that it returns the mapped port instead of the internal one.

Reply:
> This line is unnecessary, start() will already block until the waiting condition is reached.

So you mean if we had this snippet.
```
new MSSQLServerContainer<>(MSSQLServerTestImages.MSSQL_SERVER_IMAGE)
    .waitingFor(Wait.forListeningPort())
    .start()
```
The container will wait for the listening port? If that's the case then we may have another bug because that's not what occurs; MSSQLServerContainer (and other JDBCDatabaseContainer children) ignores wait strategies defined by `waitingFor` because the `waitUntilContainerStarted` override in JDBCContainer doesn't respect previously defined wait strategies. Let me know if this line of reasoning is correct and I can write up an issue for it.

Reply:
The logic is a bit different 😉 

This code
```
new MSSQLServerContainer<>(MSSQLServerTestImages.MSSQL_SERVER_IMAGE) .start()
```
should be fine, because as you mentioned, `JDBCDatabaseContainer` provides a mechanism to check the readiness in `waitUntilContainerStarted`. The inherited logic is to test for a successful SQL query via a JDBC connection. In order for this to succeed, it is technically guaranteed that the listening port became accessible beforehand (because else we could not connect to the JDBC URL of the container).

The `JDBCDatabaseContainer` are a bit _special_ in this regard, but it is the currently expected behavior.